src/Arcus.EventGrid/Contracts/CloudOrEventGridEvent.cs

```diff
@@ -0,0 +1,127 @@
+using CloudNative.CloudEvents;
+using GuardNet;
+using Microsoft.Azure.EventGrid.Models;
+
+namespace Arcus.EventGrid.Contracts
+{
+    /// <summary>
+    /// Representation of a supported event in Azure Event Grid.
+    /// This can either be an <see cref="CloudEvent"/> or <see cref="EventGridEvent"/> instance.
+    /// </summary>
+    /// <remarks>
+    ///     This model is not build for custom events.
+    ///     Create your own event by inheriting from <see cref="EventGridEvent{TData}"/>.
+    /// </remarks>
+    public sealed class CloudOrEventGridEvent : IEvent
+    {
+        private readonly CloudEvent _cloudEvent;
+        private readonly EventGridEvent _eventGridEvent;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CloudOrEventGridEvent"/> class.
+        /// </summary>
+        public CloudOrEventGridEvent(CloudEvent cloudEvent)
+        {
+            Guard.NotNull(cloudEvent, nameof(cloudEvent));
+
+            _cloudEvent = cloudEvent;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CloudOrEventGridEvent"/> class.
+        /// </summary>
+        public CloudOrEventGridEvent(EventGridEvent eventGridEvent)
+        {
+            Guard.NotNull(eventGridEvent, nameof(eventGridEvent));
+
+            _eventGridEvent = eventGridEvent;
+        }
+
+        /// <summary>
+        /// Represent this model as a <see cref="CloudEvent"/> or <c>null</c>.
+        /// </summary>
+        public CloudEvent AsCloudEvent()
+        {
+            return _cloudEvent;
+        }
+
+        /// <summary>
+        /// Represent this model as an <see cref="EventGridEvent"/> or <c>null</c>.
+        /// </summary>
+        public EventGridEvent AsEventGridEvent()
+        {
+            return _eventGridEvent;
+        }
+
+        /// <summary>
+        /// Represent this model as a <see cref="CloudEvent"/> or <c>null</c>.
+        /// </summary>
+        public static implicit operator CloudEvent(CloudOrEventGridEvent @event)
+        {
+            return @event._cloudEvent;
+        }
+
+        /// <summary>
+        /// Represent this model as an <see cref="EventGridEvent"/> or <c>null</c>.
+        /// </summary>
+        public static implicit operator EventGridEvent(CloudOrEventGridEvent @event)
+        {
+            return @event._eventGridEvent;
+        }
+
+        /// <summary>
+        /// Represent a <see cref="CloudEvent"/> as a <see cref="CloudOrEventGridEvent"/> representation.
+        /// </summary>
+        public static implicit operator CloudOrEventGridEvent(CloudEvent cloudEvent)
+        {
+            return new CloudOrEventGridEvent(cloudEvent);
+        }
+
+        /// <summary>
+        /// Represent an <see cref="EventGridEvent"/> as a <see cref="CloudOrEventGridEvent"/> representation.
+        /// </summary>
+        public static implicit operator CloudOrEventGridEvent(EventGridEvent eventGridEvent)
```

Comment:
I'm a newbie to this, what does this do?

Reply:
Makes sure that I can initialize a cloud event or event grid event directly without calling additional methods.
It's the same functionality as if you can pass a integer to a decimal without converting.

Reply:
Nice 👌